Explorer/Assets/DCL/UI/GenericContextMenu/GenericContextMenuController.cs

```diff
@@ -0,0 +1,177 @@
+        {
+            base.OnViewShow();
+            inputData.ActionOnShow?.Invoke();
+        }
+
+        private void ConfigureContextMenu()
+        {
+            float totalHeight = 0;
+
+            for (var i = 0; i < inputData.Config.contextMenuSettings.Count; i++)
+            {
+                ContextMenuControlSettings config = inputData.Config.contextMenuSettings[i];
+                object controlInitialValue = null;
+                inputData.InitialValues?.TryGetValue(config, out controlInitialValue);
+
+                GenericContextMenuComponentBase component = controlsPoolManager.GetContextMenuComponent(config, controlInitialValue, i);
+
+                component.RegisterCloseListener(TriggerContextMenuClose);
+
+                totalHeight += component!.RectTransformComponent.rect.height;
+            }
+
+            viewInstance!.ControlsLayoutGroup.spacing = inputData.Config.elementsSpacing;
+            viewInstance!.ControlsLayoutGroup.padding = inputData.Config.verticalLayoutPadding;
+            viewInstance.ControlsContainer.sizeDelta = new Vector2(inputData.Config.width,
+                totalHeight
+                + viewInstance!.ControlsLayoutGroup.padding.bottom
+                + viewInstance!.ControlsLayoutGroup.padding.top
+                + (viewInstance!.ControlsLayoutGroup.spacing * (inputData.Config.contextMenuSettings.Count - 1)));
+
+            viewInstance!.ControlsContainer.localPosition = GetControlsPosition(inputData.AnchorPosition, inputData.Config.offsetFromTarget, inputData.OverlapRect);
+        }
+
+        private Vector2 GetOffsetByDirection(ContextMenuOpenDirection direction, Vector2 offsetFromTarget)
+        {
+            return direction switch
+            {
+                ContextMenuOpenDirection.BOTTOM_RIGHT => offsetFromTarget,
+                ContextMenuOpenDirection.BOTTOM_LEFT => new Vector2(-offsetFromTarget.x - viewInstance!.ControlsContainer.rect.width, offsetFromTarget.y),
+                ContextMenuOpenDirection.TOP_RIGHT => new Vector2(offsetFromTarget.x, -offsetFromTarget.y + viewInstance!.ControlsContainer.rect.height),
+                ContextMenuOpenDirection.TOP_LEFT => new Vector2(-offsetFromTarget.x - viewInstance!.ControlsContainer.rect.width, -offsetFromTarget.y + viewInstance!.ControlsContainer.rect.height),
+                _ => Vector3.zero
+            };
+        }
+
+        private Vector3 GetControlsPosition(Vector2 anchorPosition, Vector2 offsetFromTarget, Rect? overlapRect)
+        {
+            Vector3 position = viewRectTransform.InverseTransformPoint(anchorPosition);
+            position.x += viewInstance!.ControlsContainer.rect.width / 2;
+            position.y -= viewInstance!.ControlsContainer.rect.height / 2;
+
+            Vector3 newPosition = Vector3.zero;
+            float minNonOverlappingArea = float.MaxValue;
+            foreach (ContextMenuOpenDirection enumVal in Enum.GetValues(typeof(ContextMenuOpenDirection)))
```

Comment:
Use `EnumUtils.Values<T>` for a typed array and cache it: you should not allocate a new array every time